lib/routes/account.js

```diff
@@ -508,17 +559,40 @@ module.exports = function (
           // the tokens are created verified.
           var shouldSendVerifyLoginEmail = requestHelper.wantsKeys(request) && emailRecord.emailVerified && doSigninConfirmation
           if (shouldSendVerifyLoginEmail) {
-            mailer.sendVerifyLoginEmail(
-              emailRecord,
-              tokenVerificationId,
-              userAgent.call({
-                acceptLanguage: request.app.acceptLanguage,
-                timestamp: Date.now(),
-                service: service,
-                redirectTo: redirectTo,
-                resume: resume
-              }, request.headers['user-agent'])
-            )
+            var location = null
+            var timeZone = null
+            getGeoData(ip, request)
+              .then(
+                function (loc) {
+                  location = {
+                    city: loc.city,
+                    country: loc.country
+                  }
+                  timeZone = loc.timeZone
+                })
+              .catch(
+                function (err) {
+                  // log error stuff
+                  log.error({ op: 'mailer.sendVerifyLoginEmail.1', err: err.message})
+                }
+              )
```

Comment:
if you are doing the same thing twice, just put most of this logic into `lib/geodb.js` and make sure to catch and log errors from there. this way we send the email no matter if the geodb fails or not.